_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,420 @@
+---
+layout: post
+title:  "Reassigning partitions in Apache Kafka Cluster"
+date: 2021-09-23
+author: shubham_rawat
+---
+
+As an Apache Kafka user, we sometimes have to scale up/down the number of Kafka brokers in our cluster depending on the use case. Addition of extra brokers can be an advantage to handle massive load, and we can use Cruise Control for general rebalancing in Strimzi since it allows us to automate the balancing of load across the cluster but what if we are scaling down the clusters. Let us understand this with the help of an example, suppose there are certain number of brokers in a cluster and now we want to remove a broker from the cluster. We need to make sure that the broker which is going to be removed should not have any assigned partitions. Strimzi's integration of Cruise Control currently doesn't support scaling down the cluster. This requires a tool that can assign the partitions from the broker to be removed, to the remaining brokers. The most convenient tool for this job is the Kafka reassignment partition tool: `kafka-reassign-partitions.sh`.
+
+<!--more-->
+
+## Why use the Kafka reassignment partition tool?
+
+The Kafka reassignment partition tool can help you address a variety of use cases.
+
+Some of these are listed here :-
+
+1. You can reassign the partition between the brokers anytime. For example, it can be used at times when you want to scale down the number of brokers in your cluster.
+   You can assign partitions from the broker to be scaled down to other brokers which will handle these partitions now.
+
+2. 2. With the help of this tool, you can increase the no. of partitions / replicas which can help with increasing the throughput of the topic.
+   
+##  Kafka reassignment partition tool
```

Comment:
For consideration ... Wouldn't it make more sense to have this chapterfirst before `Why use the Kafka reassignment partition tool?`?

Reply:
Yes I think that but according to Tom's comment I guess he wanted this to come first

Reply:
I have the changes though since I feel that Introduction should come first and I am now displaying the use of tool in the start of the blog

Reply:
Yeah, tbh I feel that the previous order was much better. Now we have the reader "overwhelmed" with technical details about the tool before he knows why to do it. Out of curiosity, @scholzj why did you find wrong the previous order?

Reply:
Well, I think the logical order does not make sense. You cannot really ask _Why use the Kafka reassignment partition tool?_ before actually introducing the tool and explaining what it is. I agree that describing all the different options is not needed for that. So maybe instead of simply reordering the sections they need to be changed more significantly. Or the titles might need to be changed.